bases/renku_data_services/background_jobs/core.py

```diff
@@ -290,7 +291,11 @@ async def migrate_storages_v2_to_data_connectors(config: SyncConfig) -> None:
             logger.error(f"Failed to migrate {storage.name}.")
             logger.error(err)
             failed_storages.append(str(storage.storage_id))
+            if first_error is None:
+                first_error = err
 
     logger.info(f"Migrated {len(storages_v2)-len(failed_storages)}/{len(storages_v2)} data connectors.")
     if failed_storages:
         logger.error(f"Migration failed for storages: {failed_storages}.")
+    if first_error is not None:
+        raise first_error
```

Comment:
Question: Would there be a way to raise an exception which somehow combines all exceptions which happen during the job?